src/Nexus.js

```diff
@@ -0,0 +1,175 @@
+            message: "Grade not found",
+            statusCode: 404
+        });
+    }
+};
+
+fluid.defaults("gpii.nexus.writeDefaults.handler", {
+    gradeNames: ["kettle.request.http"],
+    invokers: {
+        handleRequest: {
+            funcName: "gpii.nexus.writeDefaults.handleRequest",
+            args: ["{request}.req.params.gradeName", "{request}"]
+        }
+    }
+});
+
+gpii.nexus.writeDefaults.handleRequest = function (gradeName, request) {
+    fluid.defaults(gradeName, request.req.body);
+    request.events.onSuccess.fire();
+};
+
+fluid.defaults("gpii.nexus.constructComponent.handler", {
+    gradeNames: ["kettle.request.http"],
+    invokers: {
+        handleRequest: {
+            funcName: "gpii.nexus.constructComponent.handleRequest",
+            args: ["{request}.req.params.path", "{request}"]
+        }
+    }
+});
+
+gpii.nexus.constructComponent.handleRequest = function (path, request) {
+    fluid.construct(path, request.req.body);
+    request.events.onSuccess.fire();
+};
+
+fluid.defaults("gpii.nexus.destroyComponent.handler", {
+    gradeNames: ["kettle.request.http"],
+    invokers: {
+        handleRequest: {
+            funcName: "gpii.nexus.destroyComponent.handleRequest",
+            args: ["{request}.req.params.path", "{request}"]
+        }
+    }
+});
+
+gpii.nexus.destroyComponent.handleRequest = function (path, request) {
+    fluid.destroy(path);
+    request.events.onSuccess.fire();
+};
+
+fluid.defaults("gpii.nexus.bindModel.handler", {
+    gradeNames: ["kettle.request.ws"],
+    members: {
+        // We store the targetComponent inside a container so that the
+        // component is isolated from IoC references. This will not be
+        // necessary in the future after upcoming framework changes
+        // are completed.
+        // See https://issues.fluidproject.org/browse/FLUID-4925
+        componentHolder: {
+            targetComponent: null // Will be set at onBindWs
+        },
+        modelPath: null // Will be set at onBindWs
+    },
+    invokers: {
+        targetModelChangeListener: {
+            funcName: "gpii.nexus.bindModel.targetModelChangeListener",
+            args: [
+                "{that}",
+                "{arguments}.0" // value
+            ]
+        }
+    },
+    listeners: {
+        onBindWs: {
+            funcName: "gpii.nexus.bindModel.bindWs",
+            args: [
+                "{that}",
+                "{request}.req.params.componentPath",
+                "{request}.req.params.modelPath",
+                "{that}.targetModelChangeListener"
+            ]
+        },
+        onReceiveMessage: {
+            funcName: "gpii.nexus.bindModel.receiveMessage",
+            args: [
+                "{that}.componentHolder.targetComponent",
+                "{that}.modelPath",
+                "{arguments}.1" // message
+            ]
+        },
+        onDestroy: {
+            "this": "{that}.componentHolder.targetComponent.applier.modelChanged",
```

Comment:
Yes, this is all pretty interesting, and I think it's the best we can do with the current framework. Ideally this can be done automatically with a framework-bound model listener which is co-scoped with the component, but the current framework can't deal with asynchronous component lifecycle points (such as onCreate and onDestroy). One day it can happen.